spec/client_spec.js

```diff
@@ -357,73 +373,102 @@ describe('Client', function() {
         expect(promise).to.be.rejectedWith(Error, 'Invocation request timeout').and.notify(done);
       });
 
-      it('resolves the promise when the expected message is received', function(done) {
-        promise = subject.get('ticket.subject');
+      it('returns an error when not passing in strings', function() {
+        requestsCount--;
+        expect(function() {
+          subject.get(123);
+        }).to.throw(Error);
+
+        expect(function() {
+          subject.get({
+            'ticket.subject': true
+          });
+        }).to.throw(Error);
+      });
+    });
 
-        expect(promise).to.eventually.become({a: 'b'}).and.notify(done);
+    describe('#set', function() {
+      it('returns a promise', function(done) {
+        promise = subject.set('ticket.subject', 'value');
+
+        expect(promise).to.be.a.promise;
+        expect(promise).to.eventually.become({ errors: {}, 'ticket.subject': 'value' }).and.notify(done);
 
         window.addEventListener.callArgWith(1, {
           origin: subject._origin,
           source: subject._source,
-          data: { id: requestsCount, result: { errors: {}, 'ticket.subject': {a: 'b'} } }
+          data: { id: requestsCount, result: { errors: {}, 'ticket.subject': 'value' } }
         });
       });
 
-      it("throws an error, which you can catch, when requesting something that doesn't exist", function(done) {
-        var promise = subject.get('ticket.subj');
+      it('throws an error when not including a value', function() {
+        requestsCount--;
+        expect(function() {
+          subject.set('ticket.subject');
+        }).to.throw(Error);
+      });
+
+      it('throws an error when the handler throws it', function(done) {
+        promise = subject.set('ticket.foo', 'bar');
 
-        expect(promise).to.be.rejectedWith(Error, 'No such Api').and.notify(done);
+        expect(promise).to.be.a.promise;
+        expect(promise).to.be.rejectedWith(Error, 'ticket.foo unavailable').and.notify(done);
 
         window.addEventListener.callArgWith(1, {
           origin: subject._origin,
           source: subject._source,
-          data: { id: requestsCount, result: { errors: { 'ticket.subj': { message: 'No such Api' } } } }
+          data: { id: requestsCount, result: { errors: { 'ticket.foo': { message: 'ticket.foo unavailable' } } } }
         });
       });
 
-      it('does not throw an error when bulk requesting', function(done) {
-        var promise = subject.get(['ticket.subj']);
+      it('accepts an object with multiple paths', function(done) {
+        promise = subject.set({
+          'ticket.subject': 'value',
+          'ticket.description': 'value'
+        });
 
-        expect(promise).to.become({ errors: { 'ticket.subj': { message: 'No such Api' } } }).and.notify(done);
+        expect(promise).to.be.a.promise;
```

Comment:
isn't this implied by the next line? or is the error message really bad when it fails?